test/es-module/test-esm-loader-programmatically.mjs

```diff
@@ -0,0 +1,158 @@
+  dummyDynamicImport: module => `await import('data:text/javascript,${module}');`
+}
+
+describe('ESM: programmatically register loaders', { concurrency: true }, () => {
+  it('works with only a dummy CLI argument', async () => {
+    const { code, signal, stdout, stderr } = await spawnPromisified(execPath, [
+      ...commonArgs,
+      '--eval',
+      `import { registerLoader } from 'node:module';` +
+      commonEvals.registerLoader(fixtures.fileURLString('es-module-loaders', 'loader-resolve-passthru.mjs')) +
+      commonEvals.registerLoader(fixtures.fileURLString('es-module-loaders', 'loader-load-passthru.mjs')) +
+      commonEvals.dummyDynamicImport('console.log(%22Hello from dynamic import%22);')
+    ]);
+
+    assert.strictEqual(stderr, '');
+    assert.strictEqual(code, 0);
+    assert.strictEqual(signal, null);
+
+    const lines = stdout.split('\n');
+
+    assert.match(lines[0], /resolve passthru/);
+    assert.match(lines[1], /load passthru/);
+    assert.match(lines[2], /Hello from dynamic import/);
+
+    assert.strictEqual(lines[3], '');
+  });
+
+  it('programmatically registered loaders are appended to an existing chaining', async () => {
+    const { code, signal, stdout, stderr } = await spawnPromisified(execPath, [
+      ...commonArgs,
+      '--loader',
+      fixtures.fileURL('es-module-loaders', 'loader-resolve-passthru.mjs'),
+      '--eval',
+      `import { registerLoader } from 'node:module';` +
+      commonEvals.registerLoader(fixtures.fileURLString('es-module-loaders', 'loader-load-passthru.mjs')) +
+      commonEvals.dummyDynamicImport('console.log(%22Hello from dynamic import%22)')
+    ]);
+
+    assert.strictEqual(stderr, '');
+    assert.strictEqual(code, 0);
+    assert.strictEqual(signal, null);
+
+    const lines = stdout.split('\n');
+
+    assert.match(lines[0], /resolve passthru/);
+    assert.match(lines[1], /resolve passthru/);
+    assert.match(lines[2], /load passthru/);
+    assert.match(lines[3], /Hello from dynamic import/);
+
+    assert.strictEqual(lines[4], '');
+  });
+
+  it('works registering loaders across files', async () => {
+    const { code, signal, stdout, stderr } = await spawnPromisified(execPath, [
+      ...commonArgs,
+      '--eval',
+      commonEvals.import(fixtures.path('es-module-loaders', 'register-programmatically-loader-load.mjs')) +
+      commonEvals.import(fixtures.path('es-module-loaders', 'register-programmatically-loader-resolve.mjs')) +
+      commonEvals.dummyDynamicImport('console.log(%22Hello from dynamic import%22)')
+    ]);
+
+    assert.strictEqual(stderr, '');
+    assert.strictEqual(code, 0);
+    assert.strictEqual(signal, null);
+
+    const lines = stdout.split('\n');
+
+    assert.match(lines[0], /resolve passthru/);
+    assert.match(lines[1], /load passthru/);
+    assert.match(lines[2], /Hello from dynamic import/);
+
+    assert.strictEqual(lines[3], '');
+  });
+
+  it('works registering loaders across virtual files', async () => {
```

Comment:
This test case is not working actually, I'll check what is happening later, but if you guys have any tip let me know 🤝